lib/core/state/mongo-users-state.js

```diff
@@ -0,0 +1,44 @@
+const UserCommands = require('../commands/user-commands')
+const MongoClient = require('../clients/mongodb-client')
+const logger = require('../../helpers/logger')
+
+class MongoUsersState {
+  /**
+   *
+   * @param {Object} settings Settings
+   * @param {String} settings.users users to create
+   */
+  constructor(settings) {
+    const { connection } = settings
+    this.users = settings.users
+    // Client
+    this.client = settings.client || new MongoClient(connection.uri)
+  }
+
+  async sync() {
+    const responses = []
+    // this should sync the users
+    // for now it only creates/updates new ones
+    const userCommands = new UserCommands(this.client)
+
+    for (let i = 0; i < this.users.length; i += 1) {
+      const user = this.users[i]
+      logger.debug('checking if user exits', user.user)
+      const response = await userCommands.usersInfo(user.user)
+      logger.debug('user resp', response)
+      // If this array is filled it means the user exists
+      if (response.users.length) {
```

Comment:
This can be an early return „if(!response.users.length“

Reply:
How would that work? This is per user.

Reply:
Ah yes, I see.